### ISSUE TITLE
Insufficient space to download does to act properly

### ISSUE THREAD

**Issue by lopezvg** [i96751414/plugin.video.torrest#9]
When insufficient space is detected, torrent status should marked as "error", no just "queued", and the Buffer&Play cancelled.

Now it gets blocked at 0% download.

**Comment by lopezvg** [comment]
In addition, the users tells me that the .torrent contains several files, and although the total size of the files is 7.5GB and there is only 1.5GB. of space available, the user has selected just one file from the torrent list, which is smaller than 1.5GB.

Therefore, Torrest should verify the space available of just the file to download and not for all the files in the torrent.

**Comment by i96751414** [comment]
Yep, I have that problem myself when using multiple files torrents. The issue here is that it is only possible to check for the entire torrent space. This is the reason why I added an option to disable the space checking.

Also, at the moment, the daemon set's the torrent as paused when the check fails (making the torrent not able to download), so you should see paused (and not queued). Can you confirm that?

**Comment by lopezvg** [comment]
I see what you say, but from my point o view, for this functionality to be helpful it should meets these criteria:

- Do the space check when the file (an not the full torrent) has been selected
- I not space is available, the torrent should be moved to a new "error" state
- In case of error, the Buffer&Play should be cancelled, stating the reason

It would be best to set this functionality off until is usable.

**Comment by i96751414** [comment]
I did use anacrolix/torrent a long time ago, however the performance was way inferior at the time. Not sure on how it is at the moment, but I have no plans on changing to it.

**Comment by icarok99** [comment]
@i96751414 The addon I use uses arvidn/libtorrent It's excellent, works perfectly I highly recommend it and it will put an end to pausing to download the content.

https://github.com/arvidn/libtorrent
